### PR TITLE
Add support for data source 'google_iam_role'

### PR DESCRIPTION
**Pull request by manuelfelipe**
Add `google_iam_role` data source to be able to fetch and query for GCP IAM roles permissions.

### CODE REVIEW

File: google/data_source_google_iam_role.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+func dataSourceGoogleIamRoleRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	roleName := d.Get("name").(string)
+	role, err := config.clientIAM.Roles.Get(roleName).Do()
+	if err != nil {
+		return handleNotFoundError(err, d, fmt.Sprintf("Error reading IAM Role %s: %s", roleName, err))
+	}
+
+	d.SetId(GetResourceNameFromSelfLink(role.Name))
```
Comment (rileykarson) [inline google/data_source_google_iam_role.go]: nit: We can set `role.name` w/o running it through `GetResourceNameFromSelfLink`.

File: website/docs/r/datasource_google_iam_role.html.markdown [hunk 2]
```diff
@@ -0,0 +1,37 @@
+---
+layout: "google"
+page_title: "Google: google_iam_role"
+sidebar_current: "docs-google-datasource-folder"
```
Comment (rileykarson) [inline website/docs/r/datasource_google_iam_role.html.markdown]: This should be `docs-google-datasource-iam-role`

Comment (rileykarson) [reply website/docs/r/datasource_google_iam_role.html.markdown]: You'll also need to add this datasource to `google.erb` so it shows in the sidebar.

File: website/docs/r/datasource_google_iam_role.html.markdown [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  value = "${data.google_iam_role.roleinfo.included_permissions}"
+}
+
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` (Required) - The name of the Role to lookup.
```
Comment (rileykarson) [inline website/docs/r/datasource_google_iam_role.html.markdown]: Can you include the part about the format from here? https://cloud.google.com/iam/reference/rest/v1/roles/get

File: website/docs/r/datasource_google_iam_role.html.markdown [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+## Attributes Reference
+
+The following attributes are exported:
+
+* `name` - The resource name of the Role.
```
Comment (rileykarson) [inline website/docs/r/datasource_google_iam_role.html.markdown]: This is redundant with the argument above

File: website/docs/r/datasource_google_iam_role.html.markdown (hunk 2 above)
Comment (manuelfelipe) [reply website/docs/r/datasource_google_iam_role.html.markdown]: also just noticed that I put originally under `/r` and not under `/d` where its suppose to be. fixed now

**Comment by manuelfelipe** [comment]
hey @rileykarson, pushed the changes. ready for a second review

File: website/docs/r/datasource_google_iam_role.html.markdown (hunk 2 above)
Comment (rileykarson) [reply website/docs/r/datasource_google_iam_role.html.markdown]: Nice catch!

**Pull request merged**